ext/node/ops/crypto/mod.rs

```diff
@@ -800,6 +801,30 @@ pub fn op_node_dh_generate(
   dh_generate(prime, prime_len, generator)
 }
 
+// TODO(lev): This duplication should be avoided.
+#[op]
+pub fn op_node_dh_generate2(
+  prime: ZeroCopyBuf,
+  prime_len: usize,
+  generator: usize,
+) -> Result<(ZeroCopyBuf, ZeroCopyBuf), AnyError> {
+  dh_generate(Some(prime).as_deref(), prime_len, generator)
+}
+
+#[op]
+pub fn op_node_dh_compute_secret(
+  prime: ZeroCopyBuf,
+  private_key: ZeroCopyBuf,
+  their_public_key: ZeroCopyBuf,
+) -> Result<ZeroCopyBuf, AnyError> {
+  let pubkey: BigUint = BigUint::from_bytes_be(their_public_key.as_ref());
+  let privkey: BigUint = BigUint::from_bytes_be(private_key.as_ref());
+  let primei: BigUint = BigUint::from_bytes_be(prime.as_ref());
+  let shared_secret: BigUint = pubkey.modpow(&privkey, &primei);
```

Comment:
Question here is whether this is constant-time. In its current form this may leak information about the private key

Reply:
The implementation doesn't look constant-time for even moduli. We're using it in other places too so i guess its fine

Reply:
I think we'll live with this since we're using it in other places.